core/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -231,22 +244,39 @@ Query rangeQuery(String field, Object lowerTerm, Object upperTerm,
                 }
                 return fields;
             }
+
+            private void validateParsed(Float value) {
+                if (
+                    value.isNaN() || value.isInfinite()
+                        || value > 65504
```

Comment:
Agreed with `Math.abs`, but do not understand about 65520. As I understand all finite floats greater than 65504 will be rounded to 65504 inside `HalfFloatPoint.halfFloatToShortBits`. Am I right?